deal with Non-ASCII characters

karthigayini opened the pull request:
@rhshah please review at your convenience.

karthigayini committed:
deal with Non-ASCII characters

karthigayini committed:
use het sites

rhshah commented on biometrics/cli.py:
```diff
@@ -216,6 +216,9 @@ def get_args():
         help='''Maximum z value for the colorscale on the heatmap.''')
+    parser_genotype.add_argument(
+        '--het', type=bool,
+        help='''Use Heterozygous sites too to calculate discordance rate''')
```
Let's use: Include Hetrozygous sites along with homozygous sites when calculating discordant rate, helps specifically in cases where there are less than 100 total number of sites

rhshah commented on biometrics/extract.py:
```diff
@@ -213,6 +213,9 @@ def _pileup(self, bam, site):
                 except:
                     base_qual = 30
+                 #to deal with non-ascii characters
+                if base_qual == '#':
```
I would do any string and add a default value of 30 as done above instead of continue

rhshah commented on biometrics/genotype.py:
```diff
@@ -253,8 +254,10 @@ def compare_samples(self, samples):
         # compute discordance rate
-
-        comparisons['DiscordanceRate'] = comparisons['HomozygousMismatch'] / (comparisons['HomozygousInRef'] + EPSILON)
+        if self.het:
```
Should this be not oppsite that if het is true do the current else statement

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
Update python-package.yml

Commit:
fix dataframe row append

need to use concat instaed of append when building up dataframes for code to work with modern python / pandas.

karthigayini committed:
spaces

karthigayini committed:
Merge branch 'Non-ascii_character' of https://github.com/msk-access/biometrics into Non-ascii_character

karthigayini committed:
ascii character and using het counts

karthigayini commented:
@rhshah Thanks for the suggestions. The comments are addressed

Commit:
fix tests

buehlere reviewed (approved):
this looks good to me!

karthigayini committed:
update 3

Min phred score

rhshah commented on biometrics/genotype.py:
```diff
@@ -13,14 +13,15 @@
 class Genotyper:
 
-    def __init__(self, no_db_compare, discordance_threshold=0.05, threads=1, zmin=None, zmax=None):
+    def __init__(self, no_db_compare, discordance_threshold=0.05, threads=1, zmin=None, zmax=None,het=False):
```
include spacing before het

rhshah commented on biometrics/minor_contamination.py:
```diff
@@ -30,8 +30,7 @@ def to_dataframe(self, samples):
             }
-
-            data = data.append(row, ignore_index=True)
+            data = pd.concat([data, pd.DataFrame([row])], ignore_index=True)
```
Can you please explain why this needs to be changed

rhshah commented on biometrics/extract.py:
```diff
@@ -211,8 +209,11 @@ def _pileup(self, bam, site):
+                    #this is a character value
                 except:
-                    base_qual = 30
+                    base_qual = chr(0+33)
```
Sorry I am not sure I understand why we would do `0+33`, I believe if the base_qual is obtained using `pileupread.alignment.qual[pileupread.query_position]`  is an integer value and in certain cases, we dont get the integer value and it is assigned some kind of string. I believe we either set in those scenario the value as 30 or we calculate avg. quality of the read itself as a default value in those cases.

rhshah committed:
Update genotype.py

karthigayini committed:
using read average

Commit:
Update extract.py
